File: tests/fixtures/patterns-positive/null_object/mutB.ts
class ConsoleLogger {
  log(msg: string): void {
    console.log(msg);
  }
}

// POSITIVO (mutación B: sin Null Object, guardas nulas repetidas en 2 sitios).
function reportStart(logger: ConsoleLogger | null): void {
  if (logger !== null) {
    logger.log("start");
  }
}
function reportEnd(logger: ConsoleLogger | null): void {
  if (logger !== null) {
    logger.log("end");
  }
}

File: tests/fixtures/patterns-positive/null_object/mutB.py
class ConsoleLogger:
    def log(self, msg):
        print(msg)


# POSITIVO (mutación B: sin Null Object, guardas nulas repetidas en 2 sitios).
def report_start(logger):
    if logger is not None:
        logger.log("start")


def report_end(logger):
    if logger is not None:
        logger.log("end")
